File: Templates/Java/src/main/java/Person/Job.java
package Person;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.LocalDate;

public class Job {
    @JsonProperty("title")
    private String title;

    @JsonProperty("company")
    private Company company;

    @JsonProperty("started_at")
    private LocalDate startedAt;

    // Getters and Setters
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Company getCompany() {
        return company;
    }

    public void setCompany(Company company) {
        this.company = company;
    }

    public LocalDate getStartedAt() {
        return startedAt;
    }

    public void setStartedAt(LocalDate startedAt) {
        this.startedAt = startedAt;
    }
}

File: Templates/TypeScript/Person/job.ts
import { JsonProperty } from "json-object-mapper";
import { Company } from "./company";

export class Job {
	@JsonProperty({ required: true })
	public title: string;

	@JsonProperty({ required: true })
	public company: Company;

	@JsonProperty({ name: "started_at", required: true })
	public startedAt: Date;
}
